src/test/groovy/com/ss/mqtt/broker/test/integration/MqttBrokerTest.groovy

```diff
@@ -11,12 +12,36 @@ import java.nio.charset.StandardCharsets
 @ContextConfiguration(classes = MqttBrokerTestConfig)
 class MqttBrokerTest extends Specification {
     
+    @Autowired
+    InetSocketAddress deviceNetworkAddress
+    
+    public static final ENCODING = StandardCharsets.UTF_8
     public static final topicFilter = "topic/Filter"
-    public static final publishPayload = "publishPayload".getBytes(StandardCharsets.UTF_8)
+    public static final publishPayload = "publishPayload".getBytes(ENCODING)
     
     @Autowired
     Mqtt5AsyncClient mqttSubscriber
     
     @Autowired
     Mqtt5AsyncClient mqttPublisher
+    
+    protected buildClient(String clientId) {
+        return MqttClient.builder()
+            .identifier(clientId)
+            .serverHost(deviceNetworkAddress.getHostName())
+            .serverPort(deviceNetworkAddress.getPort())
+            .useMqttVersion5()
+            .build()
+            .toAsync()
+    }
+    
+    protected static connectWith(Mqtt5AsyncClient client, String user, String pass){
```

Comment:
I think this method should be a part of the Connection test, not a base broker test